src/Lower.cpp

```diff
@@ -432,9 +433,17 @@ Module lower(const vector<Function> &output_funcs,
     debug(2) << "Lowering after lowering unsafe promises:\n"
              << s << "\n\n";
 
+    debug(1) << "Removing dead allocations and moving loop invariant code...\n";
     s = remove_dead_allocations(s);
     s = simplify(s);
     s = hoist_loop_invariant_values(s);
+    debug(2) << "Lowering after removing dead allocations and hoisting loop invariant values:\n"
+             << s << "\n\n";
+
+    debug(1) << "Flattening nested ramps...\n";
+    s = flatten_nested_ramps(s);
+    s = simplify(s);
```

Comment:
We currently intentionally don't simplify after the last LICM pass. Please move this lowering pass up a little if it needs simplification after it.